### PR TITLE
[git-webkit] Add land command

### PR DESCRIPTION
**Pull request by JonWBedard**
#### 2c52e40530cd68d794401010e06b0fb8449a72ce
<pre>
[git-webkit] Add land command
<a href="https://bugs.webkit.org/show_bug.cgi?id=231821">https://bugs.webkit.org/show_bug.cgi?id=231821</a>
&lt;rdar://problem/84309339 &gt;

Reviewed by NOBODY (OOPS!).

* Tools/Scripts/git-webkit: WebKit still uses Svn as it&apos;s canonical source of truth.
* Tools/Scripts/libraries/webkitscmpy/setup.py: Bump version.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/__init__.py: Ditto.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/local/git.py:
(Git._to_git_ref): Convert identifiers or revisions to git refs.
(Git.checkout): Use shared _to_git_ref function.
(Git.rebase): Support rebasing identifiers.
(Git.pull): Generalize computation of the gmtoffset.
(Git.diff_lines): Output diff between two refs.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/mocks/local/git.py:
(Git.__init__): Add branch, push and diff commands.
(Git.filter_branch): Support sed.
(Git.pull): Mock `git pull`
(Git.move_branch): Mock `git rebase`.
(Git.push): Mock `git push`.
(Git.reset): Mock `git reset HEAD~#`.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/__init__.py:
(main): Add land command.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/land.py: Added.
(Land.parser):
(Land.main): Rebase the current branch against it&apos;s root, assign the reviewer, check for &apos;OOPS&apos; messages,
rebase against target branch, update target branch ref, optionally canonicalize, push (or svn dcommit) and
update and close pull-request with information on the landed commit.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/scm_base.py:
(ScmBase.gmtoffset): Add shared computation of GMT offset.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/git_unittest.py:
(TestGit.test_commits): `git log` should exclude base ref.
(TestGit.test_commits_branch): Ditto.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/land_unittest.py: Added.
(repository):
(TestLand):
(TestLandGitHub):
(TestLandBitBucket):
</pre>

### CODE REVIEW

File: Tools/Scripts/libraries/webkitscmpy/webkitscmpy/local/git.py [hunk 1]
```diff
@@ -912,3 +916,27 @@ def modified(self, staged=None):
...
+        kwargs = dict()
+        if sys.version_info >= (3, 0):
+            kwargs = dict(encoding='utf-8')
+        target = '{}..{}'.format(base, head) if head else base
+        proc = subprocess.Popen(
+            [self.executable(), 'diff', target],
+            cwd=self.root_path,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            ** kwargs
```
Comment (dewei-zhu) [inline Tools/Scripts/libraries/webkitscmpy/webkitscmpy/local/git.py]: Nit: do we expect space between ** and kwargs?

File: Tools/Scripts/libraries/webkitscmpy/webkitscmpy/local/git.py [hunk 2]
```diff
@@ -912,3 +916,27 @@ def modified(self, staged=None):
         if set(staged) - added:
             return staged
         return staged + self.modified(staged=False)
+
+    def diff_lines(self, base, head=None):
+        base = self._to_git_ref(base)
+        head = self._to_git_ref(head)
+
+        kwargs = dict()
+        if sys.version_info >= (3, 0):
```
Comment (dewei-zhu) [inline Tools/Scripts/libraries/webkitscmpy/webkitscmpy/local/git.py]: `sys.version_info >= (3, 6)`?

File: Tools/Scripts/libraries/webkitscmpy/webkitscmpy/local/git.py (hunk 1 above)
Comment (dewei-zhu) [inline Tools/Scripts/libraries/webkitscmpy/webkitscmpy/local/git.py]: Nit: no space between ** and kwargs(?)

File: Tools/Scripts/libraries/webkitscmpy/webkitscmpy/local/git.py (hunk 2 above)
Comment (dewei-zhu) [inline Tools/Scripts/libraries/webkitscmpy/webkitscmpy/local/git.py]: `sys.version_info >= (3, 6)` ?

**Review by dewei-zhu — APPROVED** [review]

**Comment by JonWBedard** [comment]
For reference, https://github.com/WebKit/WebKit/pull/32 was landed with this script.

**Comment by JonWBedard** [comment]
Landed https://commits.webkit.org/243883@main (ab900e13d201b00254abbe268c663c313e2eec32)!

**Pull request closed without merging**